core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -642,12 +643,14 @@ Builder populate() {
       defineMethod(EXP, BuiltInMethod.EXP.method, NullPolicy.STRICT);
       defineMethod(POWER, BuiltInMethod.POWER.method, NullPolicy.STRICT);
       defineMethod(ABS, BuiltInMethod.ABS.method, NullPolicy.STRICT);
-      defineMethod(LOG2, BuiltInMethod.LOG2.method, NullPolicy.STRICT);
 
       map.put(LN, new LogImplementor());
       map.put(LOG, new LogImplementor());
       map.put(LOG10, new LogImplementor());
 
```

Comment:
Can you just create an instance of a LogMsImplementor that we can reuse? I think that's more efficient.

Would you mind doing the same for LogImplementor above?

Thanks.

Reply:
Looks like there's something wrong with my approach, I'll fix it

Reply:
fix

Reply:
I meant can you create an instance of the Log implementors and then reuse them? You're instantiating each implementor several times and they're all the same so we can just reuse the same one. 

Also would you mind not marking these as resolved until they've been reviewed? It's hard to tell what comments the newest changes address. Also would you mind waiting to squash your commits until the PR has been approved?

Reply:
Yes, I did it like this. You can’t see it over there. It seems I have to submit a PR again.

Reply:
Oh very weird, they are showing up now. Sorry about that.

Reply:
It still doesn't look like you addressed the comment though so would you mind doing that? Thanks

Reply:
Do you mean to solve everything with one new LogImplementor() instead of multiple instantiations?

Reply:
```java
LogImplementor logImplementor = new LogImplementor(BuiltInMethod.LOG.method);
map.put(LN, logImplementor);
map.put(LOG, logImplementor);
map.put(LOG10, logImplementor);

LogImplementor logMysqlSparkImplementor = new LogImplementor(BuiltInMethod.LOG_MYSQL_SPARK.method);
map.put(LOG_MYSQL_SPARK, logMysqlSparkImplementor);
map.put(LOG2, logMysqlSparkImplementor);
```